File: backend/src/main/java/com/artifax_solutions/website/model/AcquireSection.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class AcquireSection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String subTitle;

    @Column(columnDefinition = "TEXT")
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @OneToMany(mappedBy = "acquireSection", cascade = CascadeType.ALL)
    private List<AcquireSectionSlide> carouselSlides;

    @OneToMany(mappedBy = "acquireSection", cascade = CascadeType.ALL)
    private List<AcquireSectionUsefulLink> usefulLinks;

    @OneToMany(mappedBy = "acquireSection", cascade = CascadeType.ALL)
    private List<AcquireSectionButton> callToActions;
}

File: frontend/src/interface/AcquireSection.ts
export interface AcquireSection {
  id: number | undefined | null;
  title: string | undefined | null;
  subTitle: string | undefined | null;
  description: string | undefined | null;
  carouselSlides: AcquireSectionSlide[] | undefined | null;
  usefulLinks?: AcquireSectionUsefulLink[] | undefined | null;
  benefitList?: CustomizeDesignServiceBenefit[] | undefined | null;
  callToActions: AcquireSectionButton[] | undefined | null;
}

export interface AcquireSectionSlide {
  id: number | undefined | null;
  imageUrl: string | undefined | null;
  altText: string | undefined | null;
  placeholder?: string | undefined | null;
}

export interface AcquireSectionUsefulLink {
  id: number | undefined | null;
  linkText: string | undefined | null;
  url: string | undefined | null;
}

export interface AcquireSectionButton {
  id: number | undefined | null;
  buttonText: string | undefined | null;
  buttonUrl: string | undefined | null;
  description?: string | undefined | null;
}

export interface CustomizeDesignServiceBenefit {
  id: number | null | undefined;
  icon: string | undefined | null;
  benefit: string | undefined | null;
}
